Zend/zend_compile.h

```diff
@@ -740,6 +740,12 @@ struct _zend_execute_data {
 
 #include "zend_globals.h"
 
+typedef enum _zend_lex_begin_state {
+	ZEND_LEX_BEGIN_STATE_SHEBANG = 0,
+	ZEND_LEX_BEGIN_STATE_INITIAL,
+	ZEND_LEX_BEGIN_STATE_ST_IN_SCRIPTING
```

Comment:
As you already use a separate enum, maybe we can give these some clearer names. I find it hard to come up with good names, but maybe `BEFORE_SHEBANG`/`OPTIONAL_SHEBANG`, `BEFORE_OPEN_TAG`, `AFTER_OPEN_TAG`?

Reply:
What do you think of these names?

```cpp
typedef enum _zend_compile_string_position {
	ZEND_COMPILE_STRING_POSITION_AT_SHEBANG = 0,
	ZEND_COMPILE_STRING_POSITION_AT_OPEN_TAG,
	ZEND_COMPILE_STRING_POSITION_AFTER_OPEN_TAG
} zend_compile_string_position;
```

Reply:
Sounds reasonable to me. I'd maybe drop the "compile_string" from the name in case we want to reuse this for other functions, e.g. `zend_compile_position` or `zend_lex_position`?